python/tank/authentication/shotgun_wrapper.py

```diff
@@ -60,6 +63,21 @@ def _call_rpc(self, *args, **kwargs):
         except AuthenticationFault:
             logger.debug("Authentication failure.")
             pass
+        except ProtocolError as e:
```

Comment:
It would be great if the Shotgun API caught the 302 error and threw the proper exception type (some kind of AuthenticationFault derived class, which we would need to handle first) instead of the caller having to know the implementation details.